util/src/main/java/io/kubernetes/client/informer/SharedInformerFactory.java

```diff
@@ -35,6 +43,17 @@ public SharedInformerFactory() {
    * @param threadPool specified thread pool
    */
   public SharedInformerFactory(ExecutorService threadPool) {
+    this(Configuration.getDefaultApiClient(), threadPool);
+  }
+
+  /**
+   * Constructor w/ api client and thread pool specified.
+   *
+   * @param client specific api client
+   * @param threadPool specified thread pool
+   */
+  public SharedInformerFactory(ApiClient client, ExecutorService threadPool) {
+    apiClient = client;
```

Comment:
>     // set read timeout zero to ensure client doesn't time out
    apiClient.getHttpClient().setReadTimeout(0, TimeUnit.MILLISECONDS);

we don't have to call this everytime creating a lister-watcher, let's move this to the constructor